### DISCUSSION TITLE
RHF with react-select. Setting default with yup & conditional select options

### DISCUSSION THREAD

**Discussion by dmoptimal** [react-hook-form/react-hook-form#8565] [Q&A]
Hi, not sure if this is the right place to post this. Please let me know if not!

I can't figure out if I'm having trouble with react-select or react-hook-form. Basically I have a react-select component called 'machineType' which can be set to 'single' or 'multiple'. When set to 'single' I need to use a flat list to populate a second dropdown called 'machine'. When set to 'multi' I need to use a grouped list to populate the second dropdown. 

Under the hood this all works using setValue (I have used a watch() to confirm that it has changed, you can see in my example) But the UI doesn't change for the grouped react-select component. 

Also the default value doesn't load for the grouped dropdown for some reason (I am using yup & yupResolver, passed to useForm) Again this does work under the hood but doesn't show up in the UI.

I have a feeling I'm not using the Controller component correctly.

Here's an example: [https://codesandbox.io/s/weathered-cache-g3b3ny?file=%2Fsrc%2FApp.js](https://codesandbox.io/s/weathered-cache-g3b3ny?file=%2Fsrc%2FApp.js)

**Comment by bluebill1049** [comment]
I would try the following

1. make a controlled component with react select inside (without react hook form) and get that working first 
2. then try to use `useController` to purely be the "spy" for the component by supply value and onChange 

similar to this codesandbox: https://codesandbox.io/s/usecontroller-own-state-wncet2?file=/src/App.tsx

let me know if this resolve the issue eventually
